src/templates/index.jsx

```diff
@@ -187,9 +189,18 @@ const Layout = React.createClass({
     },
 
     componentDidUpdate: function() {
-        const { previouslySelectedNoteId, selectedNote } = this.state;
-        if (this.detailView && selectedNote !== previouslySelectedNoteId) {
+        if (!this.detailView) {
+            return;
+        }
+        const {
+            previouslySelectedNoteId,
+            previouslySelectedNoteScrollTop,
+            selectedNote,
+        } = this.state;
+        if (selectedNote !== previouslySelectedNoteId) {
             this.detailView.scrollTop = 0;
+        } else {
+            this.detailView.scrollTop = previouslySelectedNoteScrollTop;
```

Comment:
how about we only set the scroll top once and use a ternary to assign the new value?

```js
this.detailView.scrollTop = selectedNote !== previouslySelectedNoteId
	? previouslySelectedNoteScrollTop
	: 0;
```